chore: update base image

ChiefHolland opened the pull request:
# Contributor Comments

This PR updates the Docker Alpine base image from Python 3.10 to 3.11 and reverts a nit.  

## Pull Request Checklist

Thank you for submitting a contribution to the goat!

In order to streamline the review of your contribution we ask that you review and comply with the below requirements:

- [x] Rebase your branch against the latest commit of the target branch
- [x] If you are adding a dependency, please explain how it was chosen
- [x] If manual testing is needed in order to validate the changes, provide a testing plan and the expected results
- [x] If there is an issue associated with your Pull Request, link the issue to the PR.

ChiefHolland commented on Dockerfile:
```diff
@@ -52,8 +52,8 @@ RUN pip install pipenv \
     && git clone https://github.com/pyenv/pyenv.git --depth=1 "${PYENV_ROOT}" \
-    && echo 'command -v pyenv >/dev/null || export PATH="$PYENV_ROOT/bin:$PATH"' >> ~/.bash_profile \
-    && echo 'eval "$(pyenv init -)"' >> ~/.bash_profile \
+    && echo 'command -v pyenv >/dev/null || export PATH="$PYENV_ROOT/bin:$PATH"' >> ~/.profile \
```
Reverting the nit, because I missed that the reversion wasn't committed in the last PR.

JonZeolla commented:
@ChiefHolland this is good, but can you confirm what manual testing you performed to make sure this will work in prod?  I want to make sure it works on at least one 3.9 and one 3.10 repo.

ChiefHolland commented:
Running `docker run -v $(PWD):/goat --rm <hash>` against the operations repo (3.9) and service-spma (3.10) with mypy enabled (as a linter running in the pipenv context) revealed mypy errors in both repos:

```bash
~/sr/se/o/operations> docker run -v $(PWD):/goat --rm 83364

Running Seiso Linter
--------------------------

Running linter: ruff
Running linter: hadolint
Running linter: kubeconform
Running linter: dockerfile_lint
Running linter: markdown-link-check
Running linter: cspell
Running linter: cfn-lint
Running linter: jscpd
Running linter: actionlint
Running linter: markdownlint
Running linter: textlint
Running linter: black
Running linter: mypy
Running linter: shellcheck
Running linter: yamllint
===============================
MYPY
-------------------------------
pipenv run mypy --config-file /etc/opt/goat/.mypy.ini --install-types --non-interactive ./diagram_builder/draw.py ./tasks.py
diagram_builder/draw.py:18: error: Module "diagrams" has no attribute "Diagram"  [attr-defined]
diagram_builder/draw.py:18: error: Module "diagrams" has no attribute "Edge"  [attr-defined]
Found 2 errors in 1 file (checked 2 source files)

Scanned 32 files in 16 seconds
Excluded 0 files

INFO:  ruff completed successfully
INFO:  cspell completed successfully
INFO:  black completed successfully
INFO:  dockerfile_lint completed successfully
INFO:  markdownlint completed successfully
INFO:  kubeconform completed successfully
INFO:  markdown-link-check completed successfully
INFO:  jscpd completed successfully
INFO:  yamllint completed successfully
INFO:  cfn-lint completed successfully
INFO:  hadolint completed successfully
INFO:  textlint completed successfully
INFO:  shellcheck completed successfully
INFO:  actionlint completed successfully
ERROR:  mypy found errors
ERROR:  Linters found errors
```

```bash
 ~/sr/se/service-spma> docker run -v $(PWD):/goat --rm 83364  

Running Seiso Linter
--------------------------

Running linter: ruff
Running linter: hadolint
Running linter: kubeconform
Running linter: dockerfile_lint
Running linter: markdown-link-check
Running linter: cspell
Running linter: cfn-lint
Running linter: jscpd
Running linter: actionlint
Running linter: markdownlint
Running linter: textlint
Running linter: black
Running linter: mypy
Running linter: shellcheck
Running linter: yamllint
===============================
MYPY
-------------------------------
pipenv run mypy --config-file /etc/opt/goat/.mypy.ini --install-types --non-interactive ./docassemble/SalesPackageGeneratorServiceSPMA/__init__.py ./docassemble/__init__.py ./setup.py ./tasks.py
Collecting types-requests
  Downloading types_requests-2.31.0.1-py3-none-any.whl (14 kB)
Collecting types-urllib3 (from types-requests)
  Downloading types_urllib3-1.26.25.13-py3-none-any.whl (15 kB)
Installing collected packages: types-urllib3, types-requests
Successfully installed types-requests-2.31.0.1 types-urllib3-1.26.25.13
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
tasks.py:121: error: Incompatible return value type (got "Optional[str]", expected "str")  [return-value]
tasks.py:291: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
Installing missing stub packages:
/usr/local/bin/python3 -m pip install types-requests


Found 1 error in 1 file (checked 4 source files)

Scanned 249 files in 10 seconds
Excluded 0 files

INFO:  ruff completed successfully
INFO:  yamllint completed successfully
INFO:  jscpd completed successfully
INFO:  cspell completed successfully
INFO:  markdownlint completed successfully
INFO:  cfn-lint completed successfully
INFO:  shellcheck completed successfully
INFO:  hadolint completed successfully
INFO:  markdown-link-check completed successfully
INFO:  dockerfile_lint completed successfully
INFO:  kubeconform completed successfully
INFO:  textlint completed successfully
INFO:  actionlint completed successfully
INFO:  black completed successfully
ERROR:  mypy found errors
ERROR:  Linters found errors
```

ChiefHolland commented:
Also added some test files in the operations repo to force Ruff and Black errors:

```
===============================
BLACK
-------------------------------
pipenv run black --config /etc/opt/goat/.python-black --diff --check ./diagram_builder/draw.py ./tasks.py ./test.py
--- test.py     2023-07-12 16:05:40.062645 +0000
+++ test.py     2023-07-12 16:06:16.286287 +0000
@@ -1,3 +1,3 @@
 import os
 
-print ( "test")
+print("test")
would reformat test.py

Oh no! 💥 💔 💥
1 file would be reformatted, 2 files would be left unchanged.
===============================
RUFF
-------------------------------
pipenv run ruff check --config /etc/opt/goat/pyproject.toml -v ./diagram_builder/draw.py ./tasks.py ./test.py
Warning: Python 3.9 was not found on your system...
Creating a virtualenv for this project...
Pipfile: /goat/Pipfile
Using default python from /usr/local/bin/python (3.11.4) to create virtualenv...
created virtual environment CPython3.11.4.final.0-64 in 803ms
  creator CPython3Posix(dest=/tmp/goat-iSxpmfSV, clear=False, no_vcs_ignore=False, global=False)
  seeder FromAppData(download=False, pip=bundle, setuptools=bundle, wheel=bundle, via=copy, app_data_dir=/root/.local/share/virtualenv)
    added seed packages: pip==23.1.2, setuptools==67.8.0, wheel==0.40.0
  activators BashActivator,CShellActivator,FishActivator,NushellActivator,PowerShellActivator,PythonActivator

✔ Successfully created virtual environment!
Virtualenv location: /tmp/goat-iSxpmfSV
Warning: Your Pipfile requires python_version 3.9, but you are using 3.11.4 (/tmp/g/bin/python).
  $ pipenv --rm and rebuilding the virtual environment may resolve the issue.
  $ pipenv check will surely fail.
[2023-07-12][16:06:12][ruff_cli::commands::run][DEBUG] Identified files to lint in: 19.70302ms
[2023-07-12][16:06:12][ruff_cli::diagnostics][DEBUG] Checking: /goat/test.py
[2023-07-12][16:06:12][ruff_cli::diagnostics][DEBUG] Cache hit for: /goat/tasks.py
[2023-07-12][16:06:12][ruff_cli::diagnostics][DEBUG] Cache hit for: /goat/diagram_builder/draw.py
[2023-07-12][16:06:12][ruff_cli::commands::run][DEBUG] Checked 3 files in: 33.727986ms
test.py:1:8: F401 [*] `os` imported but unused
Found 1 error.
[*] 1 potentially fixable with the --fix option.
```

ChiefHolland commented:
And the same for service-spma:

```
===============================
RUFF
-------------------------------
pipenv run ruff check --config /etc/opt/goat/pyproject.toml -v ./docassemble/SalesPackageGeneratorServiceSPMA/__init__.py ./docassemble/__init__.py ./setup.py ./tasks.py ./test.py
Warning: Python 3.10 was not found on your system...
Creating a virtualenv for this project...
Pipfile: /goat/Pipfile
Using default python from /usr/local/bin/python (3.11.4) to create virtualenv...
created virtual environment CPython3.11.4.final.0-64 in 940ms
  creator CPython3Posix(dest=/tmp/goat-iSxpmfSV, clear=False, no_vcs_ignore=False, global=False)
  seeder FromAppData(download=False, pip=bundle, setuptools=bundle, wheel=bundle, via=copy, app_data_dir=/root/.local/share/virtualenv)
    added seed packages: pip==23.1.2, setuptools==67.8.0, wheel==0.40.0
  activators BashActivator,CShellActivator,FishActivator,NushellActivator,PowerShellActivator,PythonActivator

✔ Successfully created virtual environment!
Virtualenv location: /tmp/goat-iSxpmfSV
Warning: Your Pipfile requires python_version 3.10, but you are using 3.11.4 (/tmp/g/bin/python).
  $ pipenv --rm and rebuilding the virtual environment may resolve the issue.
  $ pipenv check will surely fail.
[2023-07-12][16:10:35][ruff_cli::commands::run][DEBUG] Identified files to lint in: 15.546791ms
[2023-07-12][16:10:35][ruff_cli::diagnostics][DEBUG] Checking: /goat/test.py
[2023-07-12][16:10:35][ruff_cli::diagnostics][DEBUG] Checking: /goat/setup.py
[2023-07-12][16:10:35][ruff_cli::diagnostics][DEBUG] Checking: /goat/tasks.py
[2023-07-12][16:10:35][ruff_cli::diagnostics][DEBUG] Checking: /goat/docassemble/__init__.py
[2023-07-12][16:10:36][ruff_cli::diagnostics][DEBUG] Checking: /goat/docassemble/SalesPackageGeneratorServiceSPMA/__init__.py
[2023-07-12][16:10:36][ruff_cli::commands::run][DEBUG] Checked 5 files in: 42.900393ms
test.py:1:8: F401 [*] `os` imported but unused
Found 1 error.
[*] 1 potentially fixable with the --fix option.
```

JonZeolla commented:
LGTM!  Those mypy errors are expected and they disable mypy for the time being, due to an upstream issue.

JonZeolla reviewed (approved):
LGTM
